File: frontend/lib/models/advice.dart
class Advice {
  final String category;
  final String text;
  final int correctionLevel;

  Advice({required this.category, required this.text, required this.correctionLevel});

  factory Advice.fromJson(Map<String, dynamic> json) {
    return Advice(
      category: json['category'],
      text: json['text'],
      correctionLevel: json['correction_level'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'category': category,
      'text': text,
      'correction_level': correctionLevel,
    };
  }
}

File: backend/exercise_correction/models/advice.py
from django.db import models

from .repetition import Repetition


class Advice(models.Model):
    repetition = models.ForeignKey(Repetition, related_name='advice', on_delete=models.CASCADE)
    category = models.TextField()
    text = models.TextField()
    correction_level = models.IntegerField()

    def __str__(self):
        return f"Advice of type {self.category} for repetition {self.repetition.id} with level {self.correction_level}"
